contrib/endpoints/include/api_manager/request.h

```diff
@@ -70,6 +70,9 @@ class Request {
   // Adds a header to backend. If the header exists, overwrite its value
   virtual utils::Status AddHeaderToBackend(const std::string &key,
                                            const std::string &value) = 0;
+
+  // Gets all HTTP request headers.
+  virtual void GetHeaders(std::map<std::string, std::string> *headers) {}
```

Comment:
Envoy is not using this class.   I think it is fine to make this pure since ESP is always using SHA.   When you update proxy SHA at ESP to point to this one,  that PR should add its override too.
Only one special Jenkins test will fail.  For each PR in istio/proxy we have a Jenkins test: change proxy SHA at ESP to the new one blindly and run test.  This test will fail.  But it is fine since we know this is a broken change.